tools/validation-generator/src/main/java/io/spine/tools/validate/code/IsSet.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.validate.code;
+
+import com.google.protobuf.Descriptors.FieldDescriptor.JavaType;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeName;
+import io.spine.code.proto.FieldDeclaration;
+import io.spine.tools.validate.FieldAccess;
+import io.spine.tools.validate.MessageAccess;
+import io.spine.tools.validate.field.Containers;
+import io.spine.validate.Alternative;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static com.squareup.javapoet.ClassName.bestGuess;
+import static io.spine.tools.validate.code.BooleanExpression.falseLiteral;
+import static io.spine.tools.validate.code.BooleanExpression.trueLiteral;
+import static java.lang.String.format;
+import static javax.lang.model.element.Modifier.PRIVATE;
+import static javax.lang.model.element.Modifier.STATIC;
+
+/**
+ * A factory of code which check is a given field is set.
+ */
+public final class IsSet {
+
+    private static final String MESSAGE = "msg";
+
+    private final FieldDeclaration field;
+    private final String methodName;
+    private final CodeBlock methodBody;
+    private boolean alwaysTrue = false;
+
+    public IsSet(FieldDeclaration field) {
+        this.field = checkNotNull(field);
+        this.methodName = format("is%sSet", field.name()
+                                                 .toCamelCase());
+        this.methodBody = methodBody();
+    }
+
+    /**
+     * Invokes the method which checks if the value is set.
```

Comment:
Not quite `invokes`. I think it generates the expression, in turn, checking whether the value is set.